draft-ietf-mls-protocol.md

```diff
@@ -2570,10 +2569,14 @@ message at the same time, by taking the following steps:
   corresponds to the order of PreSharedKey proposals in the `proposals` vector.
   Otherwise, set `psk_secret` to a zero-length octet string.
 
-* Construct an MLSPlaintext object containing the Commit object. Sign the MLSPlaintext
-  using the current epoch's GroupContext as context. Use the signature, the
-  `commit_secret` and the `psk_secret` to advance the key schedule and compute
-  the `confirmation_tag` value in the MLSPlaintext.
+* Use the `commit_secret`, the `psk_secret`, the provisional GroupContext, and
+  the init secret from the previous epoch to compute the epoch secret and
+  derived secrets for the new epoch.
+
+* Construct an MLSPlaintext object containing the Commit object. Sign the
+  MLSPlaintext using the current epoch's GroupContext as context. Use the
+  `confirmation_key` for the new epoch to compute the `confirmation_tag` value
+  in the MLSPlaintext.
```

Comment:
Again, mixing epochs seems bad

Reply:
This is how it works in the "A member of the group applies a Commit message by taking the following steps:" section.  It verifies the signature before doing any modifications, and uses the `confirmation_key` from the new epoch.